### PR TITLE
Run as unprivileged user (and create that user via packages)

### PR DESCRIPTION
**Pull request by olifre**
This creates a user and group `xrootd-monitoring-shoveler` and runs the service as that user. 
Furthermore, it adds a dependency of the service on `network-online.target` to prevent it from coming up before the network is online. 

While not strictly necessary, it's likely better to merge this after https://github.com/opensciencegrid/xrootd-monitoring-shoveler/pull/34 to ensure it does not break anything, since that PR tests the packaging and execution. 
I tried that in my fork (to ensure I don't break things :wink: ), and it worked well in my tests.

### CODE REVIEW

**Comment by djw8605** [comment]
This looks good, have you tested on both debian and RPM?

**Comment by olifre** [comment]
> This looks good, have you tested on both debian and RPM?

Thanks! 

I have not tested on a physical system, but https://github.com/opensciencegrid/xrootd-monitoring-shoveler/pull/34 installs the packages directly on various distros (RockyLinux 8 and 9, Debian stable, testing and unstable). Since it uses the packages built with `goreleaser` and runs them in a matching OS container, I think that's (almost) as good as testing on a real system (I tested in a fork applying #34 first and then #47 ). 

If you prefer an actual test on a physical node, I can also do that latest next week.

File: scripts/deb-preinstall.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+SERVER_HOME=/var/spool/xrootd-monitoring-shoveler
+SERVER_USER=xrootd-monitoring-shoveler
+SERVER_GROUP=xrootd-monitoring-shoveler
+SERVER_NAME="XRootD monitoring shoveler user"
+
+# create user to avoid running server as root
+# 1. create group if not existing
+if ! getent group | grep -q "^$SERVER_GROUP:" ; then
```
Comment (matyasselmeci) [inline scripts/deb-preinstall.sh:10]:
```suggestion
if ! getent group $SERVER_GROUP >/dev/null; then
```
I tried this command out in a debian:latest container and it seemed to work (returns 0 if the group exists and 2 if it doesn't) so I don't think the grep construction is necessary.

File: scripts/deb-preinstall.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+   echo -n "Adding group $SERVER_GROUP.."
+   addgroup --quiet --system $SERVER_GROUP 2>/dev/null ||true
+   echo "..done"
+fi
+# 1. create user if not existing
+if ! getent passwd | grep -q "^$SERVER_USER:"; then
```
Comment (matyasselmeci) [inline scripts/deb-preinstall.sh:16]:
```suggestion
if ! getent passwd $SERVER_USER >/dev/null; then
```
see above

**Review by matyasselmeci — APPROVED** [review]
two suggestions but otherwise fine.

File: scripts/deb-preinstall.sh (hunk 1 above)
Comment (olifre) [reply scripts/deb-preinstall.sh:10]: Indeed, you are of course correct. I'll adapt this and force-push, always good to save and extra `grep` ;-). It also comes with the advantage that it works with non-enumerable entities. 

For reference, these commands were based on [Debian's packaging guidelines](https://www.debian.org/doc/manuals/securing-debian-manual/bpp-lower-privs.en.html) for "Securing Debian", perhaps I should have questioned these more.

Comment (matyasselmeci) [reply scripts/deb-preinstall.sh]: Heh, that doc says it's from 2017 but I wouldn't be surprised if that block of code had been copy-pasted for over 20 years...

**Pull request merged**